### PR TITLE
finish classes, clean up documentation, update prototype notebook to …

### LINKED ISSUE #65: convert spatial disagg prototype into classes
convert spatial disagg prototype notebook into classes for adding it to scikit-downscale package

### PR DESCRIPTION
**Pull request by dgergel**
This PR updates the spatial disaggregation workflow to use a new `SpatialDisaggregator` class 

 - [X] closes #65 
 - [ ] tests added / passed
 - [ ] docs reflect changes

The new `core.py` file is the crux of this PR, the added jupyter notebook just updates the spatial disagg prototype notebook to use the class contained in `core.py` versus the functions in `utils.py`. 

Note: I still have everything under the `/notebooks` subdirectory, but it probably makes sense for this repo to start having a more well-defined structure and might be useful to go ahead and incorporate that into this PR, or separate it out into a followup one.

### CODE REVIEW

**Review by kemccusker — CHANGES_REQUESTED** [review]
One change that must be addressed before merging: 
- core.py L112 -should the application of scale factor actually be: `sff_daily * climo` ?

General comments:
This PR looks super! A couple general comments that don't need to hold up the merge but would be helpful to think about:
**SpatialDisaggregator**
1. My lack of knowledge about `scikit-downscale` here: should `SpatialDisaggregator` inherit from any `scikit-downscale` base classes and use `fit` and `predict` methods? For example, `fit_scaling_factor` could just be `fit` and `apply_scaling_factor` could be `predict`.
2. Explicit error checking will be useful in these functions. For example, `raise NotImplementedError` when a var isn't "temperature" or "precipitation". Also, I think it makes sense to change the `assert`s to raising exceptions. This is because the regridding is outside of the scikit-downscale api, and therefore the user of the API will need useful exceptions and messages to know if this function fails in this way.
3. I know `SpatialDisaggregator` was modeled after classes in `scikit-downscale` that I'm less familiar with so take this for what it's worth: there might be a case to be made to store some of the intermediate objects (like the scale_factor or obs climatology?) in the class itself, so less needs to be passed back and forth from notebook. I'm not sure though, something to think about/discuss.

**workflow notebook**
1. I notice there's still a bunch of stuff being imported from `utils` - is that still necessary?
2. Would be useful in the "steps" to highlight what is (or will be) inside of `scikit-downscale` vs outside it
3. More validation to demonstrate this is doing what you think it is would be useful, or perhaps a separate validation notebook if that's better.
4. So nitpicky - `sff_lazy_compute` could just be called `sff` ?

**__init__.py**
1. includes a bunch of imports that are also in the notebook and wasn't sure this was necessary

**Comment by dgergel** [comment]
@kemccusker thanks so much for your thorough review! 

I've updated the PR as follows: 

SpatialDisaggregator class: 
- the L112 in `core.py` was a mistake - thanks for catching that! 
- updated `SpatialDisaggregator` to use `fit` and `predict` methods directly 
- now using explicit error checking when initiating the class for variable type as well as error checking when regridding failed instead of numpy's testing module 
- I don't think we can add intermediate objects to the class itself because we need to regrid them as part of the recipe. But if you can think of some other way of handling that I'm definitely open to it. 

workflow notebook: 
- removed unnecessary imports from `utils.py`
- denoted recipe vs scikit-downscale steps 
- will add a validation notebook to this PR 
- renamed `sff_lazy_compute` to `sff`

**Review by kemccusker — APPROVED** [review]

**Pull request merged**
